cli/cmd/addon_publish.go

```diff
@@ -246,3 +249,48 @@ func readManifestJSON(manifestPath string) (ManifestJSONV1, error) {
 	}
 	return manifest, nil
 }
+
+func zipAddon(addonPath, wrapDirectory, comment string) (string, string, error) {
+	s := sha256.New()
+	zipFile, err := os.CreateTemp("", "cq-addon*.zip")
```

Comment:
In theory we could just zip it into a io.Writer and upload it directly (while at the same time calculating checksums) but then the `checksum` field would need to be updated with a separate API call (or left empty if the user doesn't call `--finalize`).